compiler/modules/comm.diagnostic/diagnostic.bal

```diff
@@ -112,7 +113,43 @@ public function messageFormat(Template t) returns string {
 
 public function format(Diagnostic d) returns string[] {
     Location loc = d.location;
+    string[] lines = [];
+    string line = loc.file.filename();
     LineColumn lc = locationLineColumn(loc);
     // lineColumn returns 0-based lines currently
-    return [loc.file.filename() + ":" + lc[0].toString() + ":" + (lc[1] + 1).toString() + ": error: " + d.message];
+    line += ":" + lc[0].toString() + ":" + (lc[1] + 1).toString();
+    line += ": error: " + d.message;
+    lines.push(line);
+    Range|Position range = loc.range;
+    lines.push(...fileLines(loc.file, range));
+    return lines;
+}
+
+function fileLines(File file, Range|Position range) returns [string, string] {
+    string line = range is Position ? file.lineContent(range) : file.lineContent(range.startPos);
+    [string, string] errMessage = [];
+    errMessage[0] = line;
+    if range is Position {
+        int column = file.lineColumn(range)[1];
+        errMessage[1] = caretLine(column);
+    }
+    else {
+        var [startLine, startColumn] = file.lineColumn(range.startPos);
+        var [endLine, endColumn] = file.lineColumn(range.endPos);
+        if startLine == endLine {
+            int caretLen = endColumn - startColumn + 1;
+            errMessage[1] = caretLine(startColumn, caretLen);
+        }
+        else {
+            // multiline err
+            errMessage[1] = caretLine(startColumn, line.length() - startColumn);
+        }
+    }
+    return errMessage;
+}
+
+function caretLine(int startIndex, int len = 1) returns string {
+    string padding = "".'join(from var i in 0 ..< startIndex select " ");
```

Comment:
Create a function `stringRepeat(string, int)` and put it in comm.lib in a file string.bal.  Then merge this function into the above one.

Reply:
fixed